### DISCUSSION TITLE
Main branch now targets mill 0.10 and is binary incompatible

### DISCUSSION THREAD

**Discussion by lefou** [com-lihaoyi/mill#1376] [News]
In order to support newer Scala versions, we needed to bump some upstream dependencies, most prominently Ammonite and upickle. Because these include binary incompatible changes all commits in our default branch "main" after e3e80b3937fb903e842423ecb41ef52eff5bab61 target a mill `0.10` release.

Please note, that most external plugins built for mill 0.9 will most likely not work properly. Some external mill plugins are already cross-compiled for different mill releases and adapted to a naming pattern, which includes the mill binary platform in it's name. So look out for a `_mill0.9` in the plugin artifact name and replace it with what ever the plugin provides, eventually a `_mill0.10`.

If you had some changes to mill in mind which require some refactoring, now is a good time to open your pull request.

**Comment by joan38** [comment]
Hi, is there a plan to support Scala 3 in the Mill DSL? Were there any reason to not have a Scala 3 switch in 0.10?

**Reply by lefou** [reply]
Mill uses macros to rewrite `T { }` constructs into the underlying applicatives. As the way macros work has changed in Scala 3, we need to analyze whether a migration to Scala 3 is possible at all and if not, what parts need to be changed and what parts can simply be ported over. Also some alternaive API/DSL could be the outcome. I haven't found the time to dig into this. Any help would be much appreciated.
